p5/sketch/Skia2DRenderer/renderer2d.py

```diff
@@ -291,3 +370,183 @@ def triangle(self, *args):
         self.path.close()
 
         self.render()
+
+    def _do_fill_stroke_close(self, close_shape):
+        if close_shape:
+            self.path.close()
+        self.render()
+
+    def end_shape(self, mode, vertices, is_curve, is_bezier, is_quadratic, is_contour, shape_kind):
+
+        close_shape = mode == constants.CLOSE
+        # NOT APPENDING AGAIN
+        num_verts = len(vertices)
+        if is_curve and shape_kind is None:
+            if num_verts > 3:
+                s = 1 - self.curve_tightness
+                self.path.moveTo(vertices[1][0], vertices[1][1])
+                for i in range(1, num_verts - 2):
+                    b = []
+                    v = vertices[i]
+                    b.append([v[0], v[1]])
+                    b.append([
+                        v[0] + (s * vertices[i + 1][0] - s * vertices[i - 1][0]) / 6,
```

Comment:
Pardon me since this question may be very obvious, but where does the 6 come from in this equation?

Reply:
https://github.com/processing/p5.js/blob/9cd186349cdb55c5faf28befff9c0d4a390e02ed/src/core/p5.Renderer2D.js#L767

I was not able to find an explanation but that's how it was done in p5.js

Reply:
I see - while I think it's ok to use logic from other Processing distributions, let's add links in the code that point to where a particular piece of logic was borrowed from.